modules/nats/nats_test.go

```diff
@@ -57,3 +59,45 @@ func TestNATS(t *testing.T) {
 
 	require.Equal(t, "hello", string(msg.Data))
 }
+
+func TestNATSWithConfigFile(t *testing.T) {
+	const natsConf = `
+listen: 0.0.0.0:4222
+authorization {
+    token: "s3cr3t"
+}
+`
+	ctx := context.Background()
+
+	ctr, err := tcnats.Run(ctx, "nats:2.9", tcnats.WithConfigFile(strings.NewReader(natsConf)))
+	testcontainers.CleanupContainer(t, ctr)
+	require.NoError(t, err)
+
+	uri, err := ctr.ConnectionString(ctx)
+	require.NoError(t, err)
+
+	// connect without a correct token must fail
+	mallory, err := nats.Connect(uri, nats.Name("Mallory"), nats.Token("secret"))
+	require.Error(t, err)
+	require.ErrorIs(t, err, nats.ErrAuthorization)
+	t.Cleanup(mallory.Close)
```

Comment:
suggestion: user EqualError to avoid the double test and make the Cleanup call first before it's a null check as currently it t will never be called

Reply:
Used a direct string message, the `nats.ErrAuthorization` returned string with lower cases `nats: authorization violation`

```go
require.EqualError(t, err, "nats: Authorization Violation")
```

ff17ba124cb6cf9c72bfd04206630fd4afb6fb8a